src/vmcu/services/communications/Decoder_Data.py

```diff
@@ -0,0 +1,151 @@
+import threading
+import struct
+import re
+
+import DatagramSocket
+
+
+class Decoder:
+    def __init__(self, package_data, ds): #package_data its a array that contains string id and string with all measurements
+        self.dict_measurement_types = {} #key string id of each measurament, contains an array from types
+        self.dict_measurement_names = {} #contains array of names of each variable
+        self.dict_measurement_value = {} #contains the value of each measurement
+        self.ds = ds #datagramsocket
+        
+        for data in package_data:
+            id = data[0]
+            measurements = data[1].split(',')
+            arr_measuremets = []
+            arr_names = []
+            for measure in measurements:
+                pair = measure.split(':')
+                arr_names.append(pair[0])
+                arr_measuremets.append(pair[1])
+                self.dict_measurement_value[pair[0]] = None
+
+            self.dict_measurement_types[id] = arr_measuremets
+            self.dict_measurement_names[id] = arr_names
+
+        self.recv_packet_thread = threading.Thread(target=self._recv_packet, daemon=True)
+        self.recv_packet_running = False
+    
+    def start(self):
+        self.recv_packet_running = True
+        self.ds._running = True
+        self.recv_packet_thread.start()
+
+    def _recv_packet(self):
+        buff = bytearray()
+        id_packet = 0
+        bytes_count = 0
+        bytes_size = 0
+        wait_new_packet = True
+
+        while self.recv_packet_running:
+            raw_data = self.ds.get_packet() #get_packet must to be blocking
+            if wait_new_packet:
+                id_packet = struct.unpack('<H',raw_data[:2])[0]
+                buff += raw_data[2:]
+                
+                bytes_size = self.calculate_byte_size(self.dict_measurement_types[id_packet])
+                bytes_count += len(raw_data) - 2
+                wait_new_packet = False
+            else:
+                buff += raw_data
+                bytes_count += len(raw_data)
+
+            if bytes_count < bytes_size:
+                continue
+            else:
+                if bytes_count == bytes_size:
+                    self.deserialize(buff, id_packet)
+
+                buff.clear()
+                bytes_count = 0
+                bytes_size = 0
+                wait_new_packet = True  
```

Comment:
What happens if the bytes_count is greater than the bytes_size? This part looses packets if you have more than one in the buffer. You should add the bytes to the buffer and then do in a loop:
1. Get the ID and the packet size
2. If the buffer has enough bytes to deserialize the packet do that
3. Remove the processed data from the buffer
4. Wait for more data